File: apps/api-rs/src/models/citation.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use super::source_anchor::{CharRange, NormalizedBBox};

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct CitationAnchor {
    pub anchor_id: Option<Uuid>,
    pub parse_job_id: Option<Uuid>,
    #[serde(default)]
    pub format: String,
    #[serde(default)]
    pub kind: String,
    #[serde(default)]
    pub page: Option<i32>,
    #[serde(default)]
    pub slide: Option<i32>,
    #[serde(default)]
    pub block_ids: Vec<Uuid>,
    #[serde(default)]
    pub table_ids: Vec<Uuid>,
    #[serde(default)]
    pub char_range: Option<CharRange>,
    #[serde(default)]
    pub bbox: Option<NormalizedBBox>,
    #[serde(default)]
    pub location_status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Citation {
    pub id: Uuid,
    pub assistant_message_id: Uuid,
    pub index: i32,
    pub chunk_id: Uuid,
    pub doc_id: Uuid,
    pub doc_title: String,
    pub page_range: Vec<i32>,
    pub heading_path: Vec<String>,
    pub quote: String,
    pub score: f64,
    pub source_status: String,
    pub anchor: Option<CitationAnchor>,
}

File: apps/api-bun/src/models/citation.ts
// 移植自 apps/api-rs/src/models/citation.rs
import type { CharRange, NormalizedBBox } from './source_anchor.ts';

export interface CitationAnchor {
  anchor_id: string | null; parse_job_id: string | null;
  format: string; kind: string;
  page: number | null; slide: number | null;
  block_ids: string[]; table_ids: string[];
  char_range: CharRange | null; bbox: NormalizedBBox | null;
  location_status: string;
}
export interface Citation {
  id: string; assistant_message_id: string; index: number; chunk_id: string; doc_id: string;
  doc_title: string; page_range: number[]; heading_path: string[]; quote: string; score: number;
  source_status: string; anchor: CitationAnchor | null;
}
